### ISSUE TITLE
[Developer] Support full tablet keyboard + reduced phone keyboard template in touch layout editor

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#2656]
Per Ron Lockwood (via email):

One thing I’ve found annoying in the touch keyboard layout of Keyman developer, is that I can’t easily get a full tablet keyboard and a reduced phone keyboard. Unless I’m missing something.

I do an import from On Screen, that gives me all the letters. Now I would like to have a phone layout that has a much reduce number of keys. I add the Phone platform. I use template and select the Basic template. Great. But when I got back to my tablet it now has the reduced set of keys.

I would like the template to just apply to the platform I’m working on.

The workaround I’ve done is to copy the code for the tablet before I apply a template and then paste it back in. But this is tedious.

Follow up:

Maybe I’ve figured something out. It seems that if I pick Traditional template, I get a normal tablet layout and a phone layout with reduced number of keys. The other templates seem to change both to reduced number. Of course I’m not quite sure because I’m getting a crash when I do this every time. You should have gotten the crash report (MD: found the crash report at https://github.com/sillsdev/keyman-issues/issues/6231).

**Comment by MattGyverLee** [comment]
Yes, I realized while researching https://github.com/keymanapp/keyman/issues/4153 that the Phone/Tablet flag works for iPhone/iPad, but not for Android Phone/Tablet. Devices like the Nexus 7, available as an option in Google Chrome debugging, should be recognized as a tablet, but it is not. I'm not sure if this currently works on "real" devices.

What are the cutoffs for distinguishing a phone, tablet, and phablet? 

You should be able to test your tablet keyboard on the iPad emulation, but currently, you can only test the tablet layout on Android by tweaking the source code at runtime:

![image](https://user-images.githubusercontent.com/986339/102246498-24ce1400-3eff-11eb-92bc-5cf5a3ac9c1a.png)

The formatting on Android tablets will still be old-school until my pull request (https://github.com/keymanapp/keyman/pull/4154) is approved.

This has implications in styling, as well as key number an placement as mentioned above.

**Comment by mcdurdin** [comment]
@MattGyverLee note that that is not related to this issue, which is around the templates available for producing phone and tablet layouts within Keyman Developer itself.

**Comment by MattGyverLee** [comment]
"But when I got back to my tablet it now has the reduced set of keys."

Fair enough, I guess I read this part wrong. I can delete my comments here if you prefer.

**Comment by mcdurdin** [comment]
> I can delete my comments here if you prefer.

It's not a big deal, we've got the comment history to clarify any confusion 😉
